packages/frontend-core/src/components/grid/stores/columns.js

```diff
@@ -73,16 +79,24 @@ export const deriveStores = context => {
 }
 
 export const createActions = context => {
-  const { columns, datasource, schema } = context
+  const { columns, datasource } = context
 
   // Updates the width of all columns
   const changeAllColumnWidths = async width => {
-    const $schema = get(schema)
-    let mutations = {}
-    Object.keys($schema).forEach(field => {
-      mutations[field] = { width }
+    const $columns = get(columns)
+    $columns.forEach(column => {
+      const { related } = column
+      const mutation = { width }
+      if (!related) {
+        datasource.actions.addSchemaMutation(column.name, mutation)
+      } else {
+        datasource.actions.addSubSchemaMutation(
+          related.subField,
+          related.field,
+          mutation
+        )
```

Comment:
I've seen this in a few places, where we now need to check the type of column then call a different function to make the schema mutation. This got me thinking that it might be nice to abstract those column differences into the datasource store, and just have a single `addSchemaMutation` which internally checks the type of column and then updates either `schemaMutations` or `subSchemaMutations` as required.

It's not a change we need to make for now - just something I might look to refactor in future!

Reply:
I was thinking something similar. I can try to refactor it to make this logic abstract to the consumer! I will do it in a different PR to keep this one from growing much more